packages/curve-ui-kit/src/features/slippage-settings/ui/SlippageSettingsModal.tsx

```diff
@@ -128,9 +128,12 @@ export const SlippageSettingsModal = ({ isOpen, maxSlippage, onSave, onClose }:
    */
   const [lastError, setLastError] = useState<Error | undefined>(undefined)
 
+  // Update form values when slippage changes or modal is closed without saving and re-opened
   useEffect(() => {
-    setFormValues(initFormValues(maxSlippage))
-  }, [maxSlippage])
+    if (isOpen) {
```

Comment:
Maybe it makes more sense to do it when you close it? It might prevent a first incorrect render.
```suggestion
    if (!isOpen) {
```

Reply:
I'm not against per se, but wouldn't the first render be covered by the initial state in
```const [formValues, setFormValues] = useState(initFormValues(maxSlippage))```?

Reply:
the first render **_ever_** would be covered, but not the 2nd one (unless the component is remounted)

Reply:
Found this useful function we can use to reset the form only when the exit transition has been completed
![image](https://github.com/user-attachments/assets/f374eb0b-bdc1-444a-a2c9-7f093a5ba960)